File: src/main/scala/scalaSupport/api/OriginalApi.scala
package scalaSupport.api

object ApiStatus extends Enumeration {
  case class Val(name: String) extends super.Val(nextId, name)
  implicit def valueToApiStatus(x: Value) = x.asInstanceOf[Val]

  val OK = Val("OK")
  val Error = Val("Error")
}

case class OperationResult(status: ApiStatus.Val, message: String)

class OriginalApi {
  def expensiveInit(): OperationResult = 
    OperationResult(ApiStatus.OK, "init successful")

  def close(): OperationResult = 
    OperationResult(ApiStatus.OK, "api closed")
    
  def operationOne(param: String): OperationResult = 
    OperationResult(ApiStatus.OK, "operationOne performed on " + param)
    
  def operationTwo(param1: String, param2: Int): OperationResult = 
    OperationResult(ApiStatus.OK, "operationTwo performed on " + param1 + ' ' + param2)
}

File: src/main/java/javaSupport/api/OriginalApi.java
package javaSupport.api;

public class OriginalApi {
	public OperationResult expensiveInit() {
		return new OperationResult(ApiStatus.OK, "init successful");
	}

	public OperationResult close() {
		return new OperationResult(ApiStatus.OK, "api closed");
	}

	public OperationResult operationOne(String param) {
		return new OperationResult(ApiStatus.OK, "operationOne performed on "
				+ param);
	}

	public OperationResult operationTwo(String param1, int param2) {
		return new OperationResult(ApiStatus.OK, "operationTwo performed on "
				+ param1 + ' ' + param2);
	}
}
